build/android/src/main/java/net.minetest.minetest/TextEntryActivity.java

```diff
@@ -0,0 +1,88 @@
+package net.minetest.minetest;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.InputType;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.View.OnKeyListener;
+import android.widget.EditText;
+
+public class TextEntryActivity extends Activity {
+	public AlertDialog inputDialog;
+	public EditText inputWidget;
+
+	private final int INPUT_MULTILINE = 0;
```

Comment:
an enum would be better

http://stackoverflow.com/questions/3293020/android-how-to-put-an-enum-in-a-bundle

Bundle:

    // put
    bundle.putSerializable("key", YourEnum.TYPE1);
    
    // get 
    YourEnum yourenum = (YourEnum) bundle.get("key");

or

    // put
    intent.putExtra("key", yourEnum);
    
    // get
    yourEnum = (YourEnum) intent.getSerializableExtra("key");